src/ray/protobuf/gcs.proto

```diff
@@ -397,19 +395,21 @@ message StoredConfig {
   map<string, string> config = 1;
 }
 
-message ObjectTableDataList {
-  repeated ObjectTableData items = 1;
-}
-
 message ObjectLocationInfo {
   bytes object_id = 1;
   repeated ObjectTableData locations = 2;
+  // For objects that have been spilled to external storage, the URL from which
+  // they can be retrieved.
+  string spilled_url = 3;
 }
 
 // A notification message about one object's locations being changed.
 message ObjectLocationChange {
   bool is_add = 1;
   ObjectTableData data = 2;
+  // The object has been spilled to this URL. This should be set xor the above
```

Comment:
Seems valid for an object to be both spilled and in memory, for example after reload.

Reply:
Yes, this is just for a single notification. I'm following the existing structure where one `ObjectLocationChange` represents a single update to the object's location. The client can process multiple of these notifications at once.